centrallix-lib/tests/test_qprintf_66.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <string.h>
+#include <stdlib.h>
+#include "qprintf.h"
+#include <assert.h>
+
+long long
+test(char** tname)
+{
+    int i, rval;
+    int iter;
+    unsigned char buf[44];
+
+    *tname = "qprintf-66 %LL insertion in middle without overflow";
+    iter = 200000;
+    for(i=0;i<iter;i++)
+    {
+        buf[43] = '\n';
+        buf[42] = '\0';
+        buf[41] = 0xff;
+        buf[40] = '\0';
```

Comment:
Recommend the trailing redzone be immediately adjacent to the area of buf[] that is written to.  I also think the first byte of the redzone should be nonzero, to detect a nul (end of qprintf'ed string) being written there.